### PR TITLE
[Refactor] Standardize conversion kwargs ophys

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
Continuation of #516 for the `write_imaging` and the `ImagingInterface`.

Looking at the code for the `roiextractors.write_segmentation`, more major refactoring will be needed to make use of the new context helper.

### CODE REVIEW

File: src/nwb_conversion_tools/tools/roiextractors/roiextractors.py [hunk 1]
```diff
@@ -212,76 +213,85 @@ def get_nwb_imaging_metadata(imgextractor: ImagingExtractor):
...
-                    buffer_size=buffer_size,
-                )
-                add_epochs(imaging=imaging, nwbfile=nwbfile)
-            io.write(nwbfile)
-    else:
+    with make_or_load_nwbfile(
+        nwbfile_path=nwbfile_path, nwbfile=nwbfile, metadata=metadata, overwrite=overwrite, verbose=verbose
+    ) as nwbfile_out:
         add_devices(nwbfile=nwbfile, metadata=metadata)
-        add_two_photon_series(imaging=imaging, nwbfile=nwbfile, metadata=metadata, use_times=use_times)
```
Comment (h-mayorquin) [inline src/nwb_conversion_tools/tools/roiextractors/roiextractors.py:283]: Why are you removing `use_times`? this argument is not used anymore it seems.

**Review by h-mayorquin — APPROVED** [review]
Looks good to me, with `write_segmentation` to be updated at some other time.

File: src/nwb_conversion_tools/tools/roiextractors/roiextractors.py (hunk 1 above)
Comment (CodyCBakerPhD) [reply src/nwb_conversion_tools/tools/roiextractors/roiextractors.py:283]: Oops, good catch. Reverted.

Comment (h-mayorquin) [reply src/nwb_conversion_tools/tools/roiextractors/roiextractors.py:283]: The tests did not catch this, though? I guess that's something we are missing.

File: src/nwb_conversion_tools/tools/roiextractors/roiextractors.py [hunk 2]
```diff
@@ -212,76 +213,87 @@ def get_nwb_imaging_metadata(imgextractor: ImagingExtractor):
...
+                DeprecationWarning,
+            )
+            nwbfile_path = save_path
+
     if metadata is None:
         metadata = dict()
     if hasattr(imaging, "nwb_metadata"):
         metadata = dict_deep_update(imaging.nwb_metadata, metadata)
     metadata = dict_deep_update(get_nwb_imaging_metadata(imaging), metadata)
-    if nwbfile is None:
```
Comment (h-mayorquin) [inline src/nwb_conversion_tools/tools/roiextractors/roiextractors.py:255]: So nice that we can rid of all of this : D

**Review by h-mayorquin — APPROVED** [review]
Seems good to me!

File: src/nwb_conversion_tools/tools/roiextractors/roiextractors.py (hunk 2 above)
Comment (CodyCBakerPhD) [reply src/nwb_conversion_tools/tools/roiextractors/roiextractors.py:255]: These functions haven't been looked at in a while, so I'm sure there's a lot more that can be improved lol

**Comment by CodyCBakerPhD** [comment]
> The tests did not catch this, though? I guess that's something we are missing.

Yeah... roiextractor writing tests could very likely be heavily improved 🤔

**Pull request merged**